spec/ruby/core/module/autoload_spec.rb

```diff
@@ -210,6 +210,13 @@ class ModuleSpecs::Autoload::HClass
     ModuleSpecs::Autoload.use_ex1.should == :good
   end
 
+  it "considers an autoload constant as loaded when autoload is called for/from the current file" do
+    filename = fixture(__FILE__, "autoload_during_require_current_file.rb")
+    require filename
+
+    ScratchPad.recorded.should be_nil
+  end
```

Comment:
What would be the result of the spec with current `master`? It would return the path for `autoload?(...)`?

Reply:
Yes it returns a path, because we lock\unlock autoload constants before\after require and don't touch added during require